src/Features/Core/Portable/FindUsages/IRemoteFindUsagesService.cs

```diff
@@ -115,22 +116,35 @@ public ValueTask SetSearchTitleAsync(string title, CancellationToken cancellatio
 
         public async ValueTask OnDefinitionFoundAsync(SerializableDefinitionItem definition, CancellationToken cancellationToken)
         {
-            var id = definition.Id;
-            var rehydrated = await definition.RehydrateAsync(_solution, cancellationToken).ConfigureAwait(false);
+            try
+            {
+                var id = definition.Id;
+                var rehydrated = await definition.RehydrateAsync(_solution, cancellationToken).ConfigureAwait(false);
 
-            lock (_idToDefinition)
+                lock (_idToDefinition)
+                {
+                    _idToDefinition.Add(id, rehydrated);
+                }
+
+                await _context.OnDefinitionFoundAsync(rehydrated, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex) when (FatalError.ReportAndPropagateUnlessCanceled(ex, cancellationToken))
```

Comment:
Not sure if you want to have a comment here saying what this is to investigate, but at the same time this is probably worth permanently keeping here for future bugs down the road?